blocks/api/parser.js

```diff
@@ -181,6 +181,28 @@ export function createBlockWithFallback( name, innerHTML, attributes ) {
 		// as invalid, or future serialization attempt results in an error
 		block.originalContent = innerHTML;
 
+		// When a block is invalid, attempt to validate again using a supplied `deprecated` definition.
+		// This allows blocks to modify their attribute and markup structure without invalidating
+		// content written in previous formats.
+		if ( ! block.isValid && blockType.deprecated ) {
+			let attributesParsedWithDeprecatedVersion;
+			const hasValidOlderVersion = find( blockType.deprecated, ( oldBlockType ) => {
+				const deprecatedBlockType = {
+					...omit( blockType, [ 'attributes', 'save', 'supports' ] ), // Parsing/Serialization properties
```

Comment:
Do we need to `omit` here, or is it enough that the spread will override those properties?

Reply:
Yes, we do because some of these properties are optional and we do not want to copy them in the old block's definition.